src/irr/asset/CImageWriterOpenEXR.cpp

```diff
@@ -86,7 +86,7 @@ namespace irr
 				for (uint64_t yPos = region->imageOffset.y; yPos < region->imageOffset.y + region->imageExtent.height; ++yPos)
 					for (uint64_t xPos = region->imageOffset.x; xPos < region->imageOffset.x + region->imageExtent.width; ++xPos)
 					{
-						const uint64_t ptrStyleEndShiftToImageDataPixel = (yPos * width * availableChannels) + (xPos * availableChannels) + region->bufferOffset;
+						const uint64_t ptrStyleEndShiftToImageDataPixel = (yPos * region->imageExtent.width * availableChannels) + (xPos * availableChannels) + region->bufferOffset;
 						const uint64_t ptrStyleIlmShiftToDataChannelPixel = (yPos * width) + xPos;
```

Comment:
wrong, general equation is `((z*Height+y)*Width+x)*PixelOrBlockSize+ByteOffset`

Height and Width are not the image extent visual/gometrucal parameters, but buffer jumps:
see https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/asset/IImage.h#L452

Reply:
What did you get the general equation from? Could you give me a reference?

Reply:
Now it doesn't work

Reply:
First of all: image texel format size differs, so you cannot iterate over `reinterpret_cast<const ilmType*>(data)`. Iterate bytewise instead. Now your `ptrStyleEndShiftToImageDataPixel` includes `bufferOffset` and then used to increment ptr of type `ilmType*`. So ptr is getting moved by `sizeof(ilmType)*bufferOffset` bytes.
IDK why actually you are iterating over ilmType array since (as i said) channel size varies (half=2bytes or float=4bytes) and ilmType has constant size (cpp is statically typed), but anyway: to get ptr to texel you need to use equation matt gave you. If EXR is always 2D image (i dont know), you can simplify it to `(y*Width+x)*PixelOrBlockSize+ByteOffset`. It's theoretically the same what you have now, but yours is weird because this iterating over `ilmType*` thing. So in practice:
```
void* texelPtr = `reinterpret_cast<uint8_t*>(data) + bufferOffset + (yPos * region->imageExtent.width + xPos) * PixelOrBlockSize;`
```

Reply:
ah, ilmType is template param, ok. Anyway everything above applies

Reply:
Thank you @Crisspl! I have reorganized the code and now it works
I should have perform calculations without using bytewise shifts